Manutenção de Campi (validação e novas funcionalidades)

pedro-gandra opened the pull request:
Adicionadas funcionalidades de consulta por id e listagem dos campis.
Agora, apenas ADMINS podem inserir, alterar e deletar campis.
Ao deletar um campi, agora será verificado se existe um departamento registrado no mesmo, caso exista, o servidor irá retornar uma mensagem informando o usuário que o departamento deve ser excluído antes do campi

pedro-gandra committed:
Criação das funcionalidades de listar e consultar por Id

Criação de dois servlets, um que Consulta um campi por seu id e retorna seus dados em xml, e outro que retorna um xml com todos os campis. Criação de um pacote e classe View para auxiliar

pedro-gandra committed:
Merge remote-tracking branch 'origin/dev-s1' into feature/diario/campi

pedro-gandra committed:
Validação

Criado método para checar se o usuário é um ADMIN, caso ele queira usar funcionalidades restritas para ADMINS

pedro-gandra committed:
Checar se existe um departamento registrado naquele campi

Checa se existe um departamento registrado naquele campi ao ser solicitada sua deleção, se existir ele informa que o departamento deve ser deletado antes do campi

ppassinho commented on app/src/java/diario/campi/servlets/ListarCampi.java:
```diff
@@ -0,0 +1,51 @@
+		Headers.XMLHeaders(response);
+
+		try {
+			Statement stmt = conexao.createStatement();
```
Utilizar CampiRepository

pedro-gandra committed:
Melhoria na estrutura

Uso do CampiRepository para os métodos de listagem e consulta de campis, padronização das mensagens de erro

pedro-gandra committed:
Melhoria na estrutura 2

O método de deletarCampi em CampiRepository agora também checa se algum departamento está registrado naquele campi
